pyo3-macros-backend/src/method.rs

```diff
@@ -740,6 +744,23 @@ impl<'a> FnSpec<'a> {
 
         let deprecation = deprecate_trailing_option_default(self);
 
+        let deprecated_warning = self.deprecated.as_ref().map_or(quote!{}, |d| {
+            let message = &d.message;
+            let category = &d.category;
+            match category {
+                Some(category) => {
+                    quote! {
+                        py.import_bound("warnings")?.call_method1("warn", (#message, <#category as #pyo3_path::PyTypeInfo>::type_object_bound(py)))?;
+                    }
```

Comment:
I think we should emit calls to `PyErr::warn_bound` here, instead of manually executing Python code.